dev-libs/granite/granite-5.2.4.ebuild

```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+VALA_MIN_API_VERSION=0.40
+VALA_USE_DEPEND="valadoc"
+
+inherit meson vala xdg-utils
+
+DESCRIPTION="Elementary OS library that extends GTK+"
+HOMEPAGE="https://github.com/elementary/granite"
+SRC_URI="https://github.com/elementary/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE="doc test"
+
+BDEPEND="
+	$(vala_depend)
+	>=dev-util/meson-0.48.2
+	virtual/pkgconfig
+"
+DEPEND="
+	>=dev-libs/glib-2.50:2
+	dev-libs/libgee:0.8[introspection]
+	>=x11-libs/gtk+-3.22:3[introspection]
+"
+RDEPEND="${DEPEND}"
+
+src_prepare() {
+	# Fixes wrong valadoc bin name
+	sed -i "s/find_program('valadoc')/find_program('valadoc-$(vala_best_api_version)')/g" doc/meson.build
+
+	vala_src_prepare
```

Comment:
just in case, let vala_src_prepare be the first thing to call inside src_prepare (to make sure the above function works etc)